### PR TITLE
multiple --unwindset options

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by kroening** [comment]
A future PR may move the parsing of the comma-separated sequence closer to the front-end.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2557060b84f51b090ff07bce94fb1c08324dc44c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/136520511

File: doc/cprover-manual/cbmc-unwinding.md [hunk 1]
```diff
@@ -90,13 +90,14 @@ first use:
...
+    --unwindset L:B
 
-where `L` denotes a loop ID and `B` denotes the bound for that loop.
+where `L` denotes a loop ID and `B` denotes the bound for that loop. The
+option can be given multiple times.
 
 As an example, consider a program with two loops in the function main:
 
-    --unwindset c::main.0:10,c::main.1:20
+    --unwindset c::main.0:10 --unwindset c::main.1:20
```
Comment (peterschrammel) [inline doc/cprover-manual/cbmc-unwinding.md]: How intuitive is the c:: prefix?

File: doc/cprover-manual/cbmc-unwinding.md [hunk 2]
```diff
@@ -90,13 +90,14 @@ first use:
 to obtain a list of all loops in the program. Then identify the loops
 you need to set a separate bound for, and note their loop ID. Then use:
 
-    --unwindset L:B,L:B,...
+    --unwindset L:B
 
-where `L` denotes a loop ID and `B` denotes the bound for that loop.
+where `L` denotes a loop ID and `B` denotes the bound for that loop. The
```
Comment (peterschrammel) [inline doc/cprover-manual/cbmc-unwinding.md:95]: Description of thread_nr missing.

File: src/goto-instrument/unwindset.cpp [hunk 3]
```diff
@@ -21,43 +21,52 @@ void unwindsett::parse_unwind(const std::string &unwind)
...
-void unwindsett::parse_unwindset(const std::string &unwindset)
+void unwindsett::parse_unwindset_one_loop(std::string val)
 {
-  std::vector<std::string> unwindset_loops =
-    split_string(unwindset, ',', true, true);
+  unsigned thread_nr = 0;
+  bool thread_nr_set = false;
 
-  for(auto &val : unwindset_loops)
+  if(!val.empty() && isdigit(val[0]) && val.find(":") != std::string::npos)
```
Comment (peterschrammel) [inline src/goto-instrument/unwindset.cpp:29]: Wouldn't this also be true for `0c::main.1`?

File: src/goto-instrument/unwindset.cpp [hunk 4]
```diff
@@ -21,43 +21,52 @@ void unwindsett::parse_unwind(const std::string &unwind)
...
+      uw = unsafe_string2unsigned(uw_string);
+
+    if(thread_nr_set)
+      thread_loop_map[std::pair<irep_idt, unsigned>(id, thread_nr)] = uw;
+    else
+      loop_map[id] = uw;
+  }
+}
+
+void unwindsett::parse_unwindset(const std::string &unwindset)
```
Comment (peterschrammel) [inline src/goto-instrument/unwindset.cpp:57]: Where do we still need this? Shouldn't we deprecate that?

File: doc/cprover-manual/cbmc-unwinding.md (hunk 1 above)
Comment (kroening) [reply doc/cprover-manual/cbmc-unwinding.md]: Removed

File: doc/cprover-manual/cbmc-unwinding.md (hunk 2 above)
Comment (kroening) [reply doc/cprover-manual/cbmc-unwinding.md:95]: I would say that should go (in a separate PR) into a concurrency-related section.

File: src/goto-instrument/unwindset.cpp (hunk 3 above)
Comment (kroening) [reply src/goto-instrument/unwindset.cpp:29]: yes -- but seems unrelated?

File: src/goto-instrument/unwindset.cpp (hunk 4 above)
Comment (kroening) [reply src/goto-instrument/unwindset.cpp:57]: ok

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 42a3ed2b0a935aa8b3b16e857dd36e9a080f1b7a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/136761692

**Review by peterschrammel — APPROVED** [review]

**Comment by karkhaz** [comment]
This PR fixes #5185

**Pull request merged**